numpy/core/fromnumeric.py

```diff
@@ -3237,12 +3237,14 @@ def around(a, decimals=0, out=None):
     return _wrapfunc(a, 'round', decimals=decimals, out=out)
 
 
-def _mean_dispatcher(a, axis=None, dtype=None, out=None, keepdims=None):
+def _mean_dispatcher(a, axis=None, dtype=None, out=None, keepdims=None,
+                     where=None):
     return (a, out)
```

Comment:
@mhvk, do you remember if reductions visit `type(where).__array_ufunc__`? If so, we should include `where` here too.

Reply:
Right now, it is a bit of a mess: if `where` is a positional argument, then, yes, it is checked, while if it is a keyword argument, it is not (see #8892, second item).  At the time I noticed this, there was a bit of thought of whether one should generally accept overrides (even on, say, `axis`). I think that is still up in the air, but I think for `__array_function__` at least the idea that any array argument can override makes sense - and I'd rather adjust `__array_ufunc__` to do the same (e.g., come form of calculated mask).

Reply:
```suggestion
    return (a, where, out)
```

Reply:
@eric-wieser @mhvk thank you for the input. In this case, I must admit that I don't really understand how the dispatcher comes into play and what the effect of having `where` in the return of the dispatcher is. Could you point me to the right part of the documentation or explain it a little? I did integrate your suggested change in the commit. However just for `mean` so far, since I don't fully understand it yet. Should I also include it for `var`? (line `3486` in `fromnumeric.py`)

Reply:
Yes, please do include it for `var`.

Reply:
Done